From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Thu, 20 Aug 2020 23:30:02 -0300
Subject: [PATCH 07/17] [FIO extras] cmd: bootm: allow fit with imx_hab

IMX_HAB assumes that the only authentication method supported is via
HAB, which is now what we always want necessarily, specially since FIT
signature checks is also another valid secure boot method.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
(cherry picked from commit b951a1dd081dbeb1b110f087b6c98def97b249d3)
---
 cmd/bootm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cmd/bootm.c b/cmd/bootm.c
index 46bc44f95a7..d338e2541ab 100644
--- a/cmd/bootm.c
+++ b/cmd/bootm.c
@@ -207,6 +207,11 @@ int do_bootm(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 	case IMAGE_FORMAT_ANDROID:
 		/* Do this authentication in boota command */
 		break;
+#endif
+#if defined(CONFIG_FIT) && defined(CONFIG_FIT_SIGNATURE)
+	case IMAGE_FORMAT_FIT:
+		/* Assume authentication is done by fit signature */
+		break;
 #endif
 	default:
 		printf("Not valid image format for Authentication, Please check\n");
